apps/testingaccessibility/src/hooks/use-autocomplete.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+  AutocompleteOptions,
+  AutocompleteState,
+} from '@algolia/autocomplete-core'
+import {createAutocomplete} from '@algolia/autocomplete-core'
+import {useMemo, useState} from 'react'
+import type {Hit} from '@algolia/client-search'
+
+type AutocompleteItem = Hit<any> // TODO: type this
+
+export function useAutocomplete(props: AutocompleteOptions<AutocompleteItem>) {
```

Comment:
[this implementation example from Sarah](https://github.com/sarahdayan/twitter-autocomplete/blob/d9017d34fe48c0b1421e1dcabe23424683a07344/src/hooks/useAutocomplete.ts) helped me a ton ⭐